kernel/src/upcall.rs

```diff
@@ -83,23 +87,23 @@ pub(crate) struct Upcall {
     pub(crate) upcall_id: UpcallId,
 
     /// The application data passed by the app when `subscribe()` was called.
-    pub(crate) appdata: usize,
+    pub(crate) appdata: AppdataType,
 
     /// A pointer to the first instruction of the function in the app that
     /// corresponds to this upcall.
     ///
     /// If this value is `None`, this is a null upcall, which cannot actually be
     /// scheduled. An `Upcall` can be null when it is first created, or after an
     /// app unsubscribes from an upcall.
-    pub(crate) fn_ptr: Option<NonNull<()>>,
```

Comment:
same comment re: unified "NonNull reversion"

Reply:
I can definitely add a tracking issue for reverting all of these once CHERI support improves.

Reply:
Linked.